### PR TITLE
Addons as render function

### PR DESCRIPTION
**Pull request by neotrow**

### CODE REVIEW

File: src/lib/DatePickerInput.tsx [hunk 1]
```diff
@@ -82,27 +107,40 @@ const DatePickerInput = <T extends FieldValues>(props: DatePickerInputProps<T>)
...
+          formGroupId={formGroupId.current}
           name={name}
           id={id}
           label={label}
           labelToolTip={labelToolTip}
           addonLeft={addonLeft}
           addonRight={addonRight}
-          inputGroupStyle={inputGroupStyle}
+          addonProps={{ openDatePicker, isDisabled }}
+          inputGroupStyle={!!addonLeft || !!addonRight ? { ...inputGroupStyle, alignItems: "normal" } : inputGroupStyle}
```
Comment (neotrow) [inline src/lib/DatePickerInput.tsx:136]: @severinoAmmirati @manni497 

is it okay to always have align-items normal whenver there is an addon on the datepicker?

Comment (manni497) [reply src/lib/DatePickerInput.tsx:136]: @neotrow yes :D

**Review by manni497 — CHANGES_REQUESTED** [review]

File: src/lib/DatePickerInput.tsx [hunk 2]
```diff
@@ -82,27 +107,40 @@ const DatePickerInput = <T extends FieldValues>(props: DatePickerInputProps<T>)
     setValue(name, getConvertedDate(selectedDate));
   }, [name, selectedDate, setValue, getConvertedDate]);
 
+  const openDatePicker = useCallback(() => {
+    if (!isDisabled) {
+      internalDatePickerRef.current?.setOpen(true);
```
[the remark this answers is not in the document]
Comment (neotrow) [reply src/lib/DatePickerInput.tsx:112]: @manni497 

okay, than I will rename it to `toggleDatePicker` okay? :-)

[the remark this answers is not in the document]
Comment (manni497) [reply src/lib/DatePickerInput.tsx:112]: Of course :D

[the remark this answers is not in the document]
Comment (manni497) [reply src/lib/DatePickerInput.tsx:112]: @neotrow `internalDatePickerRef.current?.setOpen(true);` or Am I wrong?  You changed just the name?

**Review by neotob — DISMISSED** [review]

File: src/lib/DatePickerInput.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (neotrow) [reply src/lib/DatePickerInput.tsx]: i'm stupid, thanks @manni497  :D

**Review by manni497 — APPROVED** [review]

**Pull request merged**